### ISSUE TITLE
Schema.path() always returns SchemaType and not the specific type of the supplied path

### ISSUE THREAD

**Issue by satyamtg** [Automattic/mongoose#11987]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Mongoose version

6.4.0

### Node.js version

16.8.0

### MongoDB server version

5.x

### Description

When I try to run `Schema.path()` on a mongoose Schema, I always get the result as `SchemaType`. However, in the types, I see many specific classes that extend SchemaType and have extra methods such as `discriminator`. I don't know if I'm doing something wrong, but I assume that TypeScript won't throw errors if correct `SchemaType` is sent as the resultant type of `path()`

### Steps to Reproduce

Sample code to reproduce -

```
import { Schema, model, SchemaType, Document, Model, Types, SchemaTypes } from 'mongoose';

interface IUser {
	name: string;
	email: string;
	organization: Types.ObjectId;
 }
 
 const userSchema = new Schema<IUser>({
	name: { type: String, required: true },
	email: { type: String, required: true },
	organization: { type: Schema.Types.ObjectId, ref: 'Organization' }
  });

const pathResult = userSchema.path('name'); // I always get SchemaType as the result
```

**Comment by RAY-EZ** [comment]
I also found this a bit confusing at first. But mongoose defines the path with the given Predefined Types eg String, Number, ObjectID.
[mongoose.prototype.path](https://mongoosejs.com/docs/api/schema.html#schema_Schema-path) here in example 
they have passed second argument as `Number`. I also thought it should return `Number` while calling `schema.path('path')`.  Calling `schema.path('fullpath', Number)` this creates a new path with `name` instance of `SchemaNumber`. 

Anyway if you want get specific type out of SchemaType `schema.path('parent.child').instance`

**Comment by satyamtg** [comment]
@RAY-EZ can you please elaborate a bit with a code example? I do know there is a getter and a setter for `path`. However, can you explain what you mean by `fullpath` and then `parent.child`. Also, I get this TS error while using `.instance` as you suggested -
```
Property 'instance' does not exist on type 'SchemaType'.ts(2339)
```

**Comment by RAY-EZ** [comment]
@satyamtg I have tried every way to reproduce error. I think your typescript acting up [restarting ts-server might help.](https://stackoverflow.com/questions/64454845/where-is-vscodes-restart-ts-server). 

```
        ...
	name: { 
          firstName: {
               type: String, required: true 
               },
         lastName: {
         ...
  });

const pathResult = userSchema.path('name');
```
I modified your example a bit. suppose you have nested Path, like `name` has `firstName`. Now to access SchemaType
you would pass `fullpath` i.e `name.firstName`. 

workaround to satisfy typescript
`const pathResult = userSchema.path('name') as SchemaType<IUser>`
I hope it works 🙂
